File: celladapter/src/main/java/io/techery/celladapter/select/mode/MultiSelectionManager.java
package io.techery.celladapter.select.mode;

import java.util.List;

public class MultiSelectionManager extends SelectionManager {

    @Override
    public void toggleSelection(int position) {
        selectableCellAdapter.toggleSelection(position);
        selectableCellAdapter.notifyItemChanged(position);
    }

    @Override
    public void setSelection(int position, boolean isSelected) {
        selectableCellAdapter.setSelection(position, isSelected);
        selectableCellAdapter.notifyItemChanged(position);
    }

    @Override
    public boolean isSelected(int position) {
        return selectableCellAdapter.getSelectedPositions().contains(position);
    }

    public List<Integer> getSelectedPositions() {
        return selectableCellAdapter.getSelectedPositions();
    }

    public boolean isAllSelected() {
        return selectableCellAdapter.getSelectedItemCount() == selectableCellAdapter.getItemCount();
    }

    public void setSelectedPositions(List<Integer> selectionPositions) {
        selectableCellAdapter.clearSelections(false);
        for (Integer position : selectionPositions) {
            selectableCellAdapter.toggleSelection(position);
        }
        selectableCellAdapter.notifyDataSetChanged();
    }

    public void setSelectionForAll(boolean isSelected) {
        for (int position = 0; position < selectableCellAdapter.getItemCount(); position++)
            setSelection(position, isSelected);
    }
}

File: celladapter-kotlin/src/main/kotlin/io/erva/celladapter/select/mode/MultiSelectionManager.kt
package io.erva.celladapter.select.mode

class MultiSelectionManager : SelectionManager() {

    override fun toggleSelection(position: Int) {
        adapter.toggleSelection(position)
        adapter.notifyItemChanged(position)
    }

    override fun setSelection(position: Int, isSelected: Boolean) {
        adapter.setSelection(position, isSelected)
        adapter.notifyItemChanged(position)
    }

    override fun isSelected(position: Int): Boolean {
        return adapter.getSelectedPositions().contains(position)
    }

    fun getSelectedPositions(): Collection<Int> {
        return adapter.getSelectedPositions()
    }

    fun isAllSelected(): Boolean {
        return adapter.getSelectedItemCount() == adapter.itemCount
    }

    fun setSelectedPositions(selectionPositions: List<Int>) {
        adapter.clearSelections(false)
        for (position in selectionPositions) {
            adapter.toggleSelection(position)
        }
        adapter.notifyDataSetChanged()
    }

    fun setSelectionForAll(isSelected: Boolean) {
        for (position in 0..adapter.itemCount - 1)
            setSelection(position, isSelected)
    }
}